runtime/jvmti/jvmtiThread.cpp

```diff
@@ -1427,6 +1441,16 @@ jvmtiSuspendResumeCallBack(J9VMThread *vmThread, J9MM_IterateObjectDescriptor *o
 						clearHaltFlag(targetThread, J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND);
 						Trc_JVMTI_threadResumed(targetThread);
 					}
+					if (inTransition) {
+						MM_ObjectAccessBarrierAPI objectAccessBarrier = MM_ObjectAccessBarrierAPI(vmThread);
+
+						/* Increment inspector count by 1 to reverse the decrement done during suspend. */
+						I_64 vthreadInspectorCount = J9OBJECT_I64_LOAD(vmThread, vthread, vm->virtualThreadInspectorCountOffset);
+						while (!objectAccessBarrier.inlineMixedObjectCompareAndSwapU64(vmThread, vthread, vm->virtualThreadInspectorCountOffset, vthreadInspectorCount, vthreadInspectorCount + 1)) {
+							vthreadInspectorCount = J9OBJECT_I64_LOAD(vmThread, vthread, vm->virtualThreadInspectorCountOffset);
+						}
+						Assert_JVMTI_true(vthreadInspectorCount < 0);
```

Comment:
A suspended virtual thread can be resumed through `ResumeThread`. Then, `ResumeAllVirtualThread` can be invoked. We are incrementing `vthreadInspectorCount` without checking if the thread is suspended. I see a case where we can double increment `vthreadInspectorCount`. Do you notice it as well?

Reply:
yes you are right, I'll add an extra check before spinning on the CAS